client/screens/SnapScreen.js

```diff
@@ -117,143 +113,91 @@ export default function SnapScreen(props) {
-        6. PIN creation
-        */
+      .then(({data: {commonName}}) => {
+        console.log('Plant identification response', commonName)
         client
           .query({
             query: GET_PLANT_BY_COMMON_NAME,
             variables: {
-              // commonName: response.data.commonName
-              commonName: 'Poison Ivy'
+              commonName
             }
           })
-          .then(plant => {
-            delete plant.data.plant.__typename
-            plantCopy = plant.data.plant
-            dispatch(addPlant(plantCopy))
+          .then(({data: {plant}}) => {
+            dispatch(addPlant(plant))
             dispatch(setLeaves())
-            console.log('plant:', plant)
-
-            console.log('then after query', {
-              ...plant.data.plant,
-              plantId: uuid(),
-              pinId: uuid(),
-              lat: location.coords.latitude,
-              lng: location.coords.longitude
-            })
-
-            //This is all new
-
             client
               .query({
                 query: GET_USER_LEAVES,
                 variables: {
                   email: 'cc'
                 }
               })
-              .then(response => {
-                console.log('XXXXXXXXXXXgetUserLeaves', response)
+              .then(({data: {user: {leaves}}}) => {
                 UpdateUserLeaves({
                   variables: {
                     id: '5',
-                    leaves: response.data.user.leaves
+                    leaves
                   }
                 })
-                  .then(response => {
-                    console.log('YYYYYYYYYUpdateUserLeaves', response)
-                    dispatch(
-                      updateUserDataLeaves(response.data.UpdateUser.leaves)
-                    )
+                  .then(({data: {UpdateUser: {leaves}}}) => {
+                    dispatch(updateUserDataLeaves(leaves))
                   })
-                  .catch(() => {
-                    console.log('Could not update user')
+                  .catch(err => {
+                    console.error(err)
                   })
               })
 
-            //this is not new
-
             CreatePinPlant({
               variables: {
-                ...plant.data.plant,
+                ...plant,
                 isPoisonous: true,
                 plantId: uuid(),
                 pinId: uuid(),
                 lat: location.coords.latitude,
                 lng: location.coords.longitude
               }
             })
-              .then(creations => {
-                console.log('creations: ', creations)
+              .then(({data: {CreatePin: {id}}}) => {
                 AddPinPlantToUser({
                   variables: {
-                    pinId: creations.data.CreatePin.id,
-                    plantId: creations.data.CreatePlant.id,
+                    pinId: id,
+                    plantId: id,
```

Comment:
The pinId and plantId need to come from CreatePin and CreatePlant respectively. This is setting the plantId to the pinId, and it will break.

Reply:
Ouch! good catch - thanks buddy @ArtsyArtys

Reply:
fixed